### PR TITLE
policy based  routing fix for empty routes on NSC reconnect

### PR DESCRIPTION
**Pull request by ljkiraly**
Fix for issue: networkservicemesh/deployments-k8s/issues/9639

The routes set by policy based routing should not interfere.

### CODE REVIEW

**Review by denis-tingaikin — APPROVED** [review]
LGTM

**Review by bellycat77 — APPROVED** [review]

**Comment by denis-tingaikin** [comment]
@szvincze, @ljkiraly ping me when you're ready to merge and test the fix on CI ;)

**Comment by szvincze** [comment]
> @szvincze, @ljkiraly ping me when you're ready to merge and test the fix on CI ;)

Sure. Today I got the information that the stability test will take this week, thus most probably we can come back with feedback on Monday.

**Comment by szvincze** [comment]
@denis-tingaikin: I just received feedback from the test that the issue was reproduced even when this patch was included. So, do not merge it because it does not solve the problem.

**Comment by denis-tingaikin** [comment]
@ljkiraly Could you have a look?


Also we can take a look into the problem

@szvincze Could we get the steps to reproduce?

**Comment by denis-tingaikin** [comment]
@szvincze ok, we started looking into the problem. Please let us know if you get any additional info.

File: pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go [hunk 1]
```diff
@@ -295,13 +298,23 @@ func del(ctx context.Context, conn *networkservice.Connection, tableIDs *generic
...
-			for tableID := range ps {
+
+			for tableID, policy := range ps.policymap {
 				nsrtid := NewNetnsRTableNextID(netNS.UniqueId(), tableID)
 				nsRTableNextIDToConnID.Delete(*nsrtid)
-			}
-			for tableID, policy := range ps {
-				if err := delRule(ctx, netlinkHandle, policy, tableID, l.Attrs().Index); err != nil {
-					return err
+				origTID := tableID
```
Comment (dualBreath) [inline pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go:305]: [minor, typo] Probably "nsrtid" and "origTID" have the same meaning of the "tid" suffix. So using the same case lower or upper seems more proper to me

**Review by dualBreath — APPROVED** [review]
LGTM

**Pull request closed without merging**